config.go

```diff
@@ -1220,6 +1225,20 @@ func ValidateConfig(cfg Config, usageMessage string,
 		return nil, err
 	}
 
+	switch {
+	// The no seed backup and auto unlock are mutually exclusive.
+	case cfg.NoSeedBackup && cfg.WalletUnlockPasswordFile != "":
+		return nil, fmt.Errorf("cannot set noseedbackup and " +
+			"wallet-unlock-password-file at the same time")
+
+	// If a password file was specified, we need it to exist.
+	case cfg.WalletUnlockPasswordFile != "" &&
+		!lnrpc.FileExists(cfg.WalletUnlockPasswordFile):
+
+		return nil, fmt.Errorf("wallet unlock password file %s does "+
```

Comment:
Would it be possible to no-op here instead of erroring? 

I'm trying to understand how I would configure a Helm chart to work with these proposed changes. To me it seems like I'd have to do the following:

- Create chart with no `WalletUnlockPasswordFile` specified and deploy it
- Run `lndinit` script
- Edit chart to add `WalletUnlockPasswordFile`
- Redeploy

Is that correct or am I missing something?

I would think maybe we could allow the user to specify `WalletUnlockPasswordFile` to be the file location they expect the password to be at _after_ they run `lndinit`. If that were allowed, then the last 2 steps from my above list could be removed?

Reply:
The idea is to run an [init container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) for your `lnd` pod. The script in that init container would always go through the 3 steps (or 4 if you want to use a seed passphrase):
 - generate seed, store it in a k8s secret. exit with special error code (open for discussion) if seed already exists
 - generate wallet password, store it in a k8s secret, exit with error code if already exists
 - run wallet initialization with generated seed/password, exit with error code if wallet already exists
 - exit init container, allow lnd container to run
 - lnd container has the above secret mounted as files, the `--wallet-unlock-password-file` points to it

The special error/exit codes are mostly there to give the script the possibility of reacting (or just logging) differently depending on the outcome (did the command do its task or was the task already completed before?)

Reply:
Ahh thanks for explaining that. Sounds great. This would be super nice!